scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,249 @@
+    """
+    Send a PUT request to each of the files in the uuid_size_map with the updated size and fileContentType extracted
+    from Azul
+    :param uuid_metadata_map:   {file_uuid: metadata} map of all the files that need to be updated
+    :param ingest_api:          IngestApi object with headers set up
+    :param dry_run:             bool value, if True no PUT operations will be performed (But information will be logged)
+    :return not_found:          List of file metadata that was not found in ingest
+    :return already_filled:     List of file metadata that was already filled
+    """
+    not_found = []
+    already_filled = []
+    for uuid, metadata in tqdm.tqdm(uuid_metadata_map.items()):
+        print(uuid, metadata)
+        # Check file exists in Ingest (Non-organically described matrices are virtually indistinguishable via metadata)
+        try:
+            logging.info(f"Searching for file metadata for file {metadata['name']}")
+            entity = ingest_api.get_entity_by_uuid('files', uuid)
+        except HTTPError:
+            not_found.append(metadata)
+            continue
+
+        # Check entity is not already filled in - Script will only fill size and fileContentType when both are missing
+        if entity.get('size') and entity.get('fileContentType'):
+            already_filled.append(metadata)
+            continue
+
+        # Add size and fileContentType to metadata
+        entity['size'] = metadata['size']
+        data_type = f"{'application/gzip' if metadata['format'] == 'fastq.gz' else 'application/octet-stream'}; dcp-type=data"
+        entity['fileContentType'] = data_type
+
+        logging.info(f"PATCH file {entity.get('fileName')}, uuid {uuid}, with size {entity['size']} and dcp-type {entity['fileContentType']}")
+
+        if dry_run:
+            continue
+
+        # PUT the metadata back with the newly introduced values
+        ingest_api.put(entity['_links']['self']['href'], json=entity)
+        logging.info(f"PATCHED file {entity.get('fileName')}, url: {entity['_links']['self']['href']}")
+
+    return not_found, already_filled
+
+
+def file_list_to_document(path: str, file_list: list):
+    with open(path, 'w') as f:
+        json.dump(file_list, f, indent=4, separators=(", ", ": "))
+
+
+def main(project_uuid, dry_run):
+    # Set up environment variables
+    config = Config(os.environ)
+
+    # Set-up logging & environment variables
+    logging.basicConfig(filename=f'{project_uuid}.log', level=config.LOG_LEVEL, filemode='w',
+                        format='%(levelname)s: %(message)s')
+    print(f"A log of the operation will be saved in {project_uuid}.log")
+    logging.info(f"Starting script for project {project_uuid}")
+    if dry_run:
+        logging.warning("Dry-run flag ON, no patch operations will be applied")
+
+    # Confirm UUID corresponds to a published Azul project
```

Comment:
Instead of a comment, you could just name the function in a clearer way.

Reply:
it's already called "confirm_published_uuid", I think it's pretty self-explanatory? I just put a small comment for each "functional" part of the script, but I can delete it if you feel it doesn't fit there

Reply:
There is no need for a comment. Definitely not for a one line block that contains a properly named function.
Properly named functions are your comments. I will send you some links for relevant training about the topic of naming, comments etc.